sktime/forecasting/base/adapters/_fbprophet.py

```diff
@@ -32,6 +32,7 @@ def fit(self, y, X=None, fh=None, **fit_params):
         -------
         self : returns an instance of self.
         """
+        self._is_fitted = False
```

Comment:
Why do we set this at the beginning of every `fit` rather than in the constructor?

Reply:
see #941

Reply:
in terms of interface: I'd say, let's keep this in.
Then, in next PR, move individual forecaster logic to `_fit` and so on, so the initial `self._is_fitted = False` appears only in the `BaseForecaster` and nowhere else. Then, refactor so state handling is nicer (however that will look like).

Much of what you dislike is caused by ensuring downwards compatibility of the interface refactor.

Reply:
>Why do we set this at the beginning of every fit rather than in the constructor?

See answer in #941. If we set it in the constructor, it does not prevent unexpected behaviour when `fit` is called twice, e.g., in a grid search that calls `fit` twice (and/or isn't careful and makes a deepcopy first).